src/OpenLoco/src/GameCommands/Airports/RemoveAirport.cpp

```diff
@@ -0,0 +1,275 @@
+        auto maybeTown = TownManager::getClosestTownAndDensity(pos);
+        if (maybeTown && (flags & Flags::apply) != 0)
+        {
+            if ((flags & (Flags::aiAllocated | Flags::ghost)) == 0)
+            {
+                auto* town = TownManager::get(maybeTown->first);
+                town->numberOfAirports--;
+            }
+        }
+
+        // TODO: introduce struct
+        auto* var_9C = reinterpret_cast<const int8_t*>(airportObj->var_9C);
+        while (var_9C[0] != -1)
+        {
+            // 0x004937FA
+            auto offset = World::TilePos2(var_9C[2], var_9C[3]);
+            offset = Math::Vector::rotate(offset, rotation);
+
+            auto worldPos = World::Pos3(World::toWorldSpace(offset), 0);
+
+            if ((airportObj->largeTiles & (1 << var_9C[0])) != 0)
+            {
+                worldPos.x += _largeTileOffsets[rotation].x;
+                worldPos.y += _largeTileOffsets[rotation].y;
+            }
+
+            worldPos += pos;
+
+            if (!removeAirportTileElement(worldPos, airportObj, var_9C[0], flags))
+            {
+                return FAILURE;
+            }
+
+            var_9C += 4;
+        }
+
+        // 0x00493858
+        // Should we update the station meta data?
+        if ((flags & Flags::ghost) == 0 && (flags & Flags::apply) != 0)
+        {
+            auto* station = StationManager::get(stationId);
+
+            removeTileFromStationAndRecalcCargo(stationId, pos, rotation);
+            station->invalidate();
+
+            recalculateStationModes(stationId);
+            recalculateStationCenter(stationId);
+            station->updateLabel();
+            station->invalidate();
+        }
+
+        return totalCost;
+    }
+
+    // 0x00493559
+    static currency32_t removeAirport(const AirportRemovalArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3(16, 16, 0));
+
+        // Find station element on tile
+        auto* stationEl = getStationEl(args.pos);
+        if (stationEl == nullptr)
+        {
+            return FAILURE;
+        }
+
+        StationId stationId = StationId::null;
+        if ((flags & Flags::ghost) != 0)
+        {
+            return loc_49372F(stationId, *stationEl, args.pos, flags);
+        }
+
+        // 0x004935DC
+        stationId = stationEl->stationId();
+        auto* station = StationManager::get(stationId);
+
+        // Try to find airport station tile at the specified Z coordinate
+        World::StationElement* foundStationEl = nullptr;
+        World::Pos3 foundPos{};
+        for (auto i = 0U; i < station->stationTileSize; i++)
+        {
+            auto& tilePos = station->stationTiles[i];
+            if (World::heightFloor(tilePos.z) != args.pos.z)
+            {
+                continue;
+            }
+
+            // Find station element on tile
+            stationEl = getStationEl(tilePos);
```

Comment:
this would only work for one rotation. You always need to remove the direction from the tilePos.z.